lib/ASTPrinter/ASTPrinter.cpp

```diff
@@ -1,5 +1,315 @@
+        out.indent(_indent);
+        out << "-->Type: " << node->getType()->getKind() << '\n';
+    }
+
+    void visitParamDecl(ParamDecl *node)
+    {
+        out.indent(_indent - 2);
+        out << "-->" << node->getName() << " : " << node->getType()->getKind()
+            << '\n';
+    }
+
+    void visitImportDecl(ImportDecl *node)
+    {
+        out.indent(_indent - 2);
+        out << "-->Module: " << node->getImportPath().toString() << '\n';
+    }
+
+    void visitFunctionDecl(FunctionDecl *node)
+    {
+        out.indent(_indent - 2);
+        out << "-->Name: " << node->getName() << '\n';
+        out.indent(_indent - 2);
+        out << "-->Return Type: " << node->getType()->getReturnType()->getKind()
+            << '\n';
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////// EXPRESSIONS /////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @brief Visits a LiteralExpr node.
+    /// @param node The LiteralExpr node to be visited.
+    void visitLiteralExpr(LiteralExpr *node)
+    {
+        out.indent(_indent - 2);
+        out << "-->";
+        std::visit(
+            [this](auto &&val) {
+                using T = std::decay_t<decltype(val)>;
+                if constexpr (std::is_same_v<T, llvm::APInt>) {
+                    this->out << "Integer: " << val;
+                } else if constexpr (std::is_same_v<T, llvm::APFloat>) {
+                    this->out << "Float: " << val.convertToDouble();
+                } else if constexpr (std::is_same_v<T, llvm::StringRef>) {
+                    this->out << "String: \"" << val.str() << "\"";
+                } else if constexpr (std::is_same_v<T, bool>) {
+                    this->out << "Boolean: " << (val ? "true" : "false");
+                } else {
+                    this->out << "Unknown type";
+                }
+            },
+            node->getValue()
+        );
+        out << "\n";
+    }
+
+    void visitRefExpr(RefExpr *node)
+    {
+        out.indent(_indent - 2);
+        out << "-->" << "Reference to: " << node->getIdentifier() << "\n";
+    }
+
+    void visitBinaryOpExpr(BinaryOpExpr *node)
+    {
+        out.indent(_indent - 4);
+        out << "-->" << node->getOperator()
+            << " Binary Operation with:" << "\n";
+        _printDetails += 2;
+    }
+
+    void visitCastExpr(CastExpr *node)
+    {
+        out.indent(_indent - 4);
+        out << "-->Casting to " << node->getDestType()->getKind() << ":\n";
+    }
+
+    void visitStructMemberExpr(StructMemberExpr *node)
+    {
+        out.indent(_indent - 4);
+        out << "-->Member: " << node->getMemberName() << " from struct:\n";
+    }
+
+    void visitUnaryOpExpr(UnaryOpExpr *node)
+    {
+        out.indent(_indent - 4);
+        out << "-->" << node->getOperator() << " Unary Operation with:\n";
```

Comment:
everywhere, you don't need to write the kind again, you already printed that it was a UnaryOp in beforeVisit, and you don't need to say "with:" because we know it's a tree and there's already indentation and arrows that are clear enough. we don't need more unnecessary text.